### PR TITLE
Set default ring in spk

### LINKED ISSUE #972: [spk] Set default ring
**As a:**
spk user
**I want:**
to be able to set my default ring
**So that:**
I can point which ring is my "master" ring/branch
**Describe the solution you'd like:**
a new `spk ring set-default <ring-name>` command to set the default ring for a spk managed project.

**Acceptance Criteria:**
- [ ] Create a new command `spk ring set-default <ring-name>`
- [ ] Command sets the default the ring to the `bedrock.yaml`

**Does this require updates to documentation?:**
Yes

### PR DESCRIPTION
**Pull request by edaena**
Closes [bedrock 942](https://github.com/microsoft/bedrock/issues/972)

### CODE REVIEW

File: src/lib/bedrockYaml.ts [hunk 1]
```diff
@@ -105,6 +105,53 @@ export const addNewService = (
   fs.writeFileSync(path.join(absPath, YAML_NAME), asYaml);
 };
 
+/**
+ * Saves the bedrock.yaml file
+ * @param dir @param dir Directory where <code>bedrock.yaml</code> file resides.
+ * @param data The data for bedrock.yaml
+ */
+export const save = (dir: string, data: IBedrockFile) => {
```
Comment (evanlouie) [inline src/lib/bedrockYaml.ts]: Use this function instead: https://github.com/CatalystCode/spk/blob/master/src/lib/bedrockYaml.ts#L26

**Review by evanlouie — CHANGES_REQUESTED** [review]

File: src/commands/ring/set-default.ts [hunk 2]
```diff
@@ -25,13 +30,14 @@ export const execute = async (
...
     checkDependencies(projectPath);
 
-    // Check if ring exists in bedrock.yaml, if not, warn and exit.
-    // Check if ring is already default, if so, warn and exit.
-    // Set ring as default in bedrock.yaml
+    // Load bedrock.yaml
+    let bedrockFile: IBedrockFile | undefined;
+
+    // Get bedrock.yaml
+    bedrockFile = loadBedrockFile(projectPath);
```
Comment (dennisseah) [inline src/commands/ring/set-default.ts]: above few lines can be 
```
const bedrockFile = loadBedrockFile(projectPath);
```

File: src/lib/bedrockYaml.ts [hunk 3]
```diff
@@ -105,6 +105,53 @@ export const addNewService = (
   fs.writeFileSync(path.join(absPath, YAML_NAME), asYaml);
 };
 
+/**
+ * Saves the bedrock.yaml file
+ * @param dir @param dir Directory where <code>bedrock.yaml</code> file resides.
```
Comment (dennisseah) [inline src/lib/bedrockYaml.ts]: duplicate `@param dir`

File: src/lib/bedrockYaml.ts [hunk 4]
```diff
@@ -105,6 +105,53 @@ export const addNewService = (
...
+ * @param ringName The name of the ring
+ */
+export const setDefaultRing = (
+  bedrockFile: IBedrockFile,
+  ringName: string,
+  path: string
+): void => {
+  const rings = Object.keys(bedrockFile.rings);
+  if (!rings.includes(ringName)) {
+    throw new Error("The ring '" + ringName + "' is not defined bedrock.yaml");
```
Comment (dennisseah) [inline src/lib/bedrockYaml.ts]: nitpicking:
```
throw new Error(`The ring '${ringName}' is not defined in ${YAML_NAME}`);
```

**Review by evanlouie — APPROVED** [review]

**Review by dennisseah — APPROVED** [review]

**Review by mtarng — APPROVED** [review]

**Pull request merged**